src/dirs.proj

```diff
@@ -15,6 +15,10 @@
     <Project Include="shims\ApiCompat.proj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(DirectoryToBuild)' != ''">
+    <Project Include="$(DirectoryToBuild)/**/*.csproj" />
```

Comment:
- Make it build in the correct sequence: always
- patterns should match those of the traversal build: [ref](https://github.com/dotnet/corefx/blob/5e324e4e11bb4f5b5a750b26f43384681ed759dd/src/ref.builds#L10), [src](https://github.com/dotnet/corefx/blob/a4ca868738d0bdbedbdf7fc929f8a4e11c3eed3e/src/src.builds#L5-L7), [package](https://github.com/dotnet/corefx/blob/a4ca868738d0bdbedbdf7fc929f8a4e11c3eed3e/src/packages.builds#L13-L15), [test](https://github.com/dotnet/corefx/blob/35735e24a65e4ed7513b6f40177056c3b35e2be4/src/tests.builds#L35) including extensions and exclusions so things aren't different for this build compared to root build
- support '-allConfigurations' including package build.  

That way it could be "one stop shopping" for someone doing work on a single library.  The way I've seen this done in our previous repos is to pass a filter down to wildcards and traversing the same set of projects rather than just re-basing the build directory and using a simple "build" on a set of projects.

That said, one could argue that the thing I am describing is a different thing: an filtered build.  What you've enabled is just a shortcut for calling build on a handful of projects.  If it were a filtered build I'd probably use it, but since its just a shortcut I'll pass.  That's just my opinion.  I'm also probably not the 90% user of our repo so you shouldn't be trying to tailor this to my preferences 😉